common/lib/xmodule/xmodule/js/src/video/02_html5_hls_video.js

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable no-console, no-param-reassign */
+/**
+ * HTML5 video player module to support HLS video playback.
+ *
+ */
+
+(function(requirejs, require, define) {
+    'use strict';
+    define('video/02_html5_hls_video.js', ['video/02_html5_video.js', 'hls', 'underscore'],
+    function(HTML5Video, HLS, _) {
+        var HLSVideo = {};
+
+        HLSVideo.Player = (function() {
+            function PlayerHLS(el, config) {
+                var self = this;
+
+                // do common initialization independent of player type
+                this.init(el, config);
+
+                // If we have only HLS sources and browser doesn't support HLS
+                // then show error message.
+                if (config.HLSOnlySources && !config.canPlayHLS) {
+                    this.showErrorMessage();
+                    return;
+                }
+
+                // Safari has native support to play HLS videos
+                if (config.browserIsSafari) {
+                    this.videoEl.attr('src', config.videoSources[0]);
```

Comment:
videoSources[0] do we have only one source at this point ? what if we don't have 0th index.